src/main/kotlin/org/malachite/estella/DemoApplication.kt

```diff
@@ -27,24 +28,51 @@ class DemoApplication {
         @Autowired userRepository: HibernateUserRepository,
         @Autowired offerRepository: HibernateOfferRepository,
         @Autowired recruitmentProcessRepository: HibernateRecruitmentProcessRepository,
-        @Autowired desiredSkillRepository: HibernateDesiredSkillRepository
+        @Autowired desiredSkillRepository: HibernateDesiredSkillRepository,
+        @Value("\${should_fake_load}") shouldFakeLoad: Boolean
     ): CommandLineRunner {
         return CommandLineRunner {
-            if (offerRepository.findAll().count() == 0) {
-                val companies = FakeOrganizations.getCompanies(FakeUsers.organizationUsers).map { organizationRepository.save(it) }
+            if (offerRepository.findAll().count() == 0 && shouldFakeLoad) {
+                loadData(
+                    organizationRepository,
+                    hrPartnerRepository,
+                    jobSeekerRepository,
+                    offerRepository,
+                    recruitmentProcessRepository,
+                    desiredSkillRepository
+                )
+            }
+        }
+    }
+}
 
-                jobSeekerRepository.saveAll(FakeLoader.getFakeJobSeekers())
+fun loadData(
+    organizationRepository: HibernateOrganizationRepository,
+    hrPartnerRepository: HibernateHrPartnerRepository,
+    jobSeekerRepository: HibernateJobSeekerRepository,
+    offerRepository: HibernateOfferRepository,
+    recruitmentProcessRepository: HibernateRecruitmentProcessRepository,
+    desiredSkillRepository: HibernateDesiredSkillRepository
+) {
+    val organizationUsers = FakeUsers.organizationUsers
+        .map { it.copy(id = null) }
+        .also { it.map { it.password = "a" } }
+    val companies = FakeOrganizations.getCompanies(organizationUsers).map { organizationRepository.save(it) }
 
-                val hrPartners = FakeLoader.getHrPartners(companies)
-                hrPartnerRepository.saveAll(hrPartners)
-                val desiredSkills = FakeDesiredSkills.desiredSkills.map { desiredSkillRepository.save(it) }
-                val offers = FakeOffers.getOffers(hrPartners, desiredSkills).map { offerRepository.save(it) }
+    FakeLoader.getFakeJobSeekers()
+        .map { it.copy(id = null, user = it.user.copy(id = null)) }
```

Comment:
Ogólnie jak się okazało, hibernate zmienia te klasy podczas zapisywania. Po pierwszym zapisaniu już tam było jakieś nie nullowe id i bardzo hibernate tego nie lubi, więc rzucał wyjątkami na prawo i lewo. Dlatego też musiałem przemappować (one przy wstawaniu aplikacji są w sumie i tak nullem, więc nic to nie zmienia)

Reply:
Yyyy jakim zapisaniu hibernate?

Reply:
podczas wywoływania save()